src/main/proto/netflix/titus/titus_job_api.proto

```diff
@@ -83,6 +83,22 @@ message Image {
   string digest = 3;
 }
 
+// Definition of a request to add a platform sidecar alongside a task
+// Note that this is *not* a user-defined sidecar, that is why it just has a
+// name. These platform-sidecars are attached a task start time, and the
+// definition of what the sidecar is is not baked into the job itself, just the
+// intent.
+message PlatformSidecar {
+  // (Required) Name of the platform sidecar requested
+  string name = 1;
+
+  // (Optional) Channel representing a pointer to releases of the sidecar
+  string channel = 2;
+
+  // (Optional) Arguments, KV pairs for configuring the sidecar
+  map<string, string> arguments = 3;
```

Comment:
I'd like to keep `arguments` as open-ended as we can, so that it can better support more robust and explicit mutations. If the contract is only for string values, then things like lists of strings and unitless numbers are very likely to be encoded as strings. That means that Sidecar mutations will be parsing strings in order to extract number values or to split a long string into a list, which is probably undesirable -- it's kind of slow, its cumbersome to do in CUE, and its highly prone to human error due to copy-pasted string parsing code.

It seems like a good fit for the [Struct](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Struct) type, although I don't know how feasible it is to pull that in here. I'm not really looking to support nested structs or anything too complex, just to enable the most common patterns like lists of strings, bools, and numbers.

My primary concern is that by enforcing an arbitrary type in the Job proto, we'll end up with workarounds that are hard to control. Being flexible early on in this design will make maintenance easier down the road, imho, and should help keep Sidecar mutations simple.

Reply:
Hmmmm. 
This struct type looks like a cool trick. Totally feasible to put it here.
Will still need to be jsonified into a annotation for the string. Looks like I can do:
https://stackoverflow.com/questions/38406211/how-to-convert-from-json-to-protobuf/58198700#58198700

And then the webhook would get real json on the back, which is probably what you want.
Otherwise I could ... base64 encode this field and then the webhook base64 decode -> proto.unmarshal ?

Reply:
Oh wow, a JSON blob for the webhook would be perfect! I don't mind b64'd protos also, but I do prefer that JSON is readable in a pinch.